src/app/pages/RewardPage/hooks/useGetFeesEarnedClaimAmount.ts

```diff
@@ -1,17 +1,142 @@
-import { useAccount } from 'app/hooks/useAccount';
-import { useCacheCallWithValue } from 'app/hooks/useCacheCallWithValue';
+import { useEffect, useState, useMemo, useCallback } from 'react';
+import { useAccount, useBlockSync } from 'app/hooks/useAccount';
+import { bridgeNetwork } from 'app/pages/BridgeDepositPage/utils/bridge-network';
+import { Asset, Chain } from 'types';
 import { getContract } from 'utils/blockchain/contract-helpers';
+import { useSelector } from 'react-redux';
+import { selectWalletProvider } from 'app/containers/WalletProvider/selectors';
+import { fixNumber } from 'utils/helpers';
+import { bignumber } from 'mathjs';
+import { AssetsDictionary } from 'utils/dictionaries/assets-dictionary';
+export interface IEarnedFee {
+  asset: Asset;
+  contract: string;
+  value: string;
+  rbtcValue: string;
+}
 
 export const useGetFeesEarnedClaimAmount = () => {
+  const { assetRates, assetRatesLoading, assetRatesLoaded } = useSelector(
+    selectWalletProvider,
+  );
+
+  const { loading, earnedFees } = useGetFeesEarned();
+
+  const calculatedFees = useMemo(
+    () =>
+      earnedFees.map(fee => {
+        const item = assetRates.find(
+          item => item.source === fee.asset && item.target === Asset.RBTC,
+        );
```

Comment:
```suggestion
        const rate = fixNumber(assetRates.find(
          item => item.source === fee.asset && item.target === Asset.RBTC,
        ));
```

Reply:
`fixNumber` method is dealing with edge cases and in case that rate is `undefined`, it will return `0`. No need for line `31`.